### PR TITLE
feat: Add workflow to mark and close stale issues

### PR DESCRIPTION
**Pull request by xpuska513**
# What
* Add workflow that mark old issues as stale and then automatically closes it after 5 days(works for PRs too)

### CODE REVIEW

File: .github/workflows/mark-and-close-stale-issues.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v5
+        with:
+          stale-issue-message: 'This issue was marked as stale and going to be closed soon'
+          stale-pr-message: 'This PR was marked as stale and going to be closed soon'
+          days-before-stale: 15
```
Comment (supertopher) [inline .github/workflows/mark-and-close-stale-issues.yml:14]: I left this feedback on the issue, but can we start with 30 plx?

File: .github/workflows/mark-and-close-stale-issues.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    - cron: '30 1 * * *'
+
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v5
+        with:
+          stale-issue-message: 'This issue was marked as stale and going to be closed soon'
+          stale-pr-message: 'This PR was marked as stale and going to be closed soon'
```
Comment (supertopher) [inline .github/workflows/mark-and-close-stale-issues.yml:13]: open to different wording, but I'd like to communicate how the user can continue.  For airbyte employees I don't really care too much.  They are adults who can ask for code reviews.  My target audience here is more an OSS contributor who is giving us their time and energy for free.  How do we clearly tell them why we are closing the issue and what they can do if they want to continue
```suggestion
          stale-pr-message: 'This PR was marked as stale due to inactivity and going to be closed soon.  You may re-open the PR if you still want to merge it.  Please consider tagging another reviewer if you are having trouble getting a review'
```

File: .github/workflows/mark-and-close-stale-issues.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+          days-before-stale: 15
+          days-before-close: 5
+          exempt-issue-labels: "needs-triage"
+          exempt-pr-labels: "do-not-stale"
```
Comment (supertopher) [inline .github/workflows/mark-and-close-stale-issues.yml:17]:
```suggestion
          exempt-pr-labels: "do-not-stale"

```

**Comment by supertopher** [comment]
I think Davin has feelings.  this works for me but I want a wider net.  I also want to reach out to the CAT team who deals with the OSS community more heavily than we do

@davinchia

File: .github/workflows/mark-and-close-stale-issues.yml (hunk 1 above)
Comment (xpuska513) [reply .github/workflows/mark-and-close-stale-issues.yml:14]: yep, no problem, that's why I added this parameter in the first place

**Comment by supertopher** [comment]
talked to the CAT team (who does alot with OSS) they are cool with us killing old issues

File: .github/workflows/mark-and-close-stale-issues.yml (hunk 1 above)
Comment (xpuska513) [reply .github/workflows/mark-and-close-stale-issues.yml]: done

**Comment by supertopher** [comment]
@git-phu let's figure out if we can agree on timelines for this

**Comment by davinchia** [comment]
Can we put this on pause for now? I think we need to figure out better heuristics for when to close issues before we go ahead with doing so.

File: .github/workflows/mark-and-close-stale-issues.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v5
+        with:
+          stale-issue-message: 'This issue was marked as stale and going to be closed soon'
+          stale-pr-message: 'This PR was marked as stale due to inactivity and going to be closed soon.  You may re-open the PR if you still want to merge it.  Please consider tagging another reviewer if you are having trouble getting a review'
+          days-before-stale: 30
+          days-before-close: 7
+          exempt-issue-labels: "needs-triage"
```
Comment (supertopher) [inline .github/workflows/mark-and-close-stale-issues.yml:16]: from a dev leads meeting, maybe add `epic` and think about other labels that may to so important we wouldn't want to delete them

Comment (supertopher) [reply .github/workflows/mark-and-close-stale-issues.yml:16]: the result from the leads meeting was `¯\_(ツ)_/¯`

**Pull request closed without merging**